test/test_modules.py

```diff
@@ -514,6 +514,109 @@ def _to_device(obj):
+            if isinstance(output, torch.Tensor):
+                d = output.dim()
+                if (d == 4 and ((input_mem_format == torch.channels_last and (not is_conv or cudnn_greater_7_6))
+                                or (module_mem_format == torch.channels_last and is_conv and cudnn_greater_7_6))):
+                    self.assertTrue(output.is_contiguous(memory_format=torch.channels_last))
+                elif (d == 5 and ((input_mem_format == torch.channels_last_3d and (not is_conv or cudnn_greater_8))
+                                  or (module_mem_format == torch.channels_last_3d and is_conv and cudnn_greater_8))):
+                    self.assertTrue(output.is_contiguous(memory_format=torch.channels_last_3d))
+                else:
+                    self.assertTrue(output.is_contiguous())
+            elif isinstance(output, (tuple, list)):
+                [_check_out_mem_format(o, input_mem_format, module_mem_format) for o in output]
+            else:
+                return
+
+        for module_input in module_inputs:
+            if module_input.forward_input is None:
+                continue
+
+            channels_last = _check_dims(module_input.forward_input.args, 4)
+            channels_last_3d = _check_dims(module_input.forward_input.args, 5)
+            input_mem_formats, module_mem_formats = _get_mem_formats(channels_last, channels_last_3d)
+
+            with freeze_rng_state():
+                # === Instantiate the module. ===
+                args, kwargs = module_input.constructor_input.args, module_input.constructor_input.kwargs
+
+                m = module_cls(*args, **kwargs)
+                m.to(device).to(dtype)
+
+                for input_mem_format in input_mem_formats:
+                    # === Change memformat of input. ===
+                    module_input.forward_input.args = _to_mem_format(module_input.forward_input.args, input_mem_format)
+                    module_input.forward_input.kwargs = _to_mem_format(module_input.forward_input.kwargs,
+                                                                       input_mem_format)
+
+                    for module_mem_format in module_mem_formats:
+                        # === Change memformat of module ===
+                        m.to(memory_format=module_mem_format)
+
+                        # === Do forward pass. ===
+                        args, kwargs = module_input.forward_input.args, module_input.forward_input.kwargs
+                        outputs = m(*args, **kwargs)
+
+                        # === Compare outputs to a reference if one is specified. ===
+                        # TODO: Handle precision
+                        reference_fn = module_input.reference_fn
+                        if reference_fn is not None:
+                            ref_outputs = reference_fn(m, *args, **kwargs)
+                            self.assertEqual(outputs, ref_outputs)
```

Comment:
As we discussed, it may be better to compare the outputs against the standard contiguous forward() outputs